File: src/main/java/jb/data/SelectedTuple.java
package jb.data;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class SelectedTuple {

    /**
     * indices[numberOfSpaceParts][numberOfSelectedClassifiers]
     */
    private int[][] indices;
    /**
     * weights[numberOfSpaceParts][numberOfSelectedClassifiers]
     */
    private double[][] weights;

}

File: src/main/kotlin/jb/data/SelectedTuple.kt
package jb.data

data class SelectedTuple(
        /**
         * indices[numberOfSpaceParts][numberOfSelectedClassifiers]
         */
        var indices: Array<IntArray>,
        /**
         * weights[numberOfSpaceParts][numberOfSelectedClassifiers]
         */
        var weights: Array<DoubleArray>)
